### DISCUSSION TITLE
Issue with downloading Radio data on Chirp.

### DISCUSSION THREAD

**Discussion by robpick** [egzumer/uvk5-chirp-driver#19] [Q&A] [answered]
Hi I have the Egzumer v0.21 flashed to my radio (Quansheng UVK5) and that all seems perfect. My problem is with connecting to Chirp.  This is what I'm trying:

- I have enabled dev mode on Chirp.
- Turned on the radio in programing mode then the programming cable.
- Loaded in the latest chirp driver module into Chirp.
- Selected download radio from the menu.
- Chosen UK-V5 egzumer from the dropdown and hit Ok

Then nothing seems to happen, I get no errors from the GUI but if I run chirp from the command line I get this.

[r04@ccl5600 ~]$ chirp --version
CHIRP next-20231223 on Linux ccl5600 6.1.69-1-lts #1 SMP PREEMPT_DYNAMIC Wed, 20 Dec 2023 17:23:40 +0000 x86_64 (Python 3.11.6)

[r04@ccl5600 ~]$ chirp -v
INFO: Python/3.11.6 // Linux/Linux-6.1.69-1-lts-x86_64-with-glibc2.38 // CHIRP/next-20231223 // wx/4.2.1 gtk3 (phoenix) wxWidgets 3.2.2.1
INFO: Server reports next-20231223 is latest
13:48:26: Debug: window wxMenuBar@0x55f35fcce660 ("menubar") lost focus even though it didn't have it
INFO: driver re-registration enabled
INFO: Loading module /home/r04/Radio/UVK5/uvk5_egzumer.py SHA256 c8bbfc5ccd1bc7e5f7a16d7eebfad18d55827e8fef3138d18bcd4be2d217e798
INFO: Registered Quansheng_UV-K5_egzumer = UVK5Radio
13:48:34: Debug: window wxMenuBar@0x55f35fcce660 ("menubar") lost focus even though it didn't have it
INFO: Found firmware: 
WARNING: Failed to backup <chirp.loaded.uvk5_egzumer.UVK5Radio object at 0x7f1a9aae9810>: 'bool' object has no attribute 'get_byte_compatible'
Traceback (most recent call last):
  File "/usr/lib/python3.11/site-packages/chirp/wxui/main.py", line 1496, in _menu_download
    editorset = ChirpEditorSet(radio, None, self._editors)
                ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/lib/python3.11/site-packages/chirp/wxui/main.py", line 173, in __init__
    features = radio.get_features()
               ^^^^^^^^^^^^^^^^^^^^
  File "/home/r04/Radio/UVK5/uvk5_egzumer.py", line 804, in get_features
    rf.valid_special_chans = self.Get_VFO_CHANNEL_NAMES()
                             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/r04/Radio/UVK5/uvk5_egzumer.py", line 749, in Get_VFO_CHANNEL_NAMES
    BANDS = BANDS_STANDARD if not isWide else BANDS_WIDE
            ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/lib/python3.11/site-packages/chirp/bitwise.py", line 499, in __bool__
    return self.get_value() != 0
           ^^^^^^^^^^^^^^^^
  File "/usr/lib/python3.11/site-packages/chirp/bitwise.py", line 709, in get_value
    data = self._subgen(self._data, self._offset).get_value()
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/lib/python3.11/site-packages/chirp/bitwise.py", line 219, in get_value
    value = self._data[self._offset:self._offset + self._size]
            ~~~~~~~~~~^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
TypeError: 'bool' object is not subscriptable

**Comment by robpick** [comment]
Please let me know if you need any more details

Thanks, Rob

**Comment by egzumer** [comment, marked as answer]
Programing mode is for FW upload ONLY! Turn the radio on like you always do.

**Reply by robpick** [reply]
God damn it, I should have known I was doing something stupid. Thanks
